### PR TITLE
py-bluepysnap: Add 2.0.1 (keep 1.0.7), update/add the dependencies

### PR DESCRIPTION
**Pull request by joni-herttuainen**
@genric : 
Note that there are few [commits](https://github.com/BlueBrain/snap/compare/v2.0.0...master) since release fixing a few small issues as well a few [pending PRs](https://github.com/BlueBrain/snap/pulls). 

If you experience other issues, please let me know, or feel free to open a ticket.

### CODE REVIEW

**Comment by joni-herttuainen** [comment]
@genric @jdcourcol :
Since this ended up being way bigger than expected, I'm not sure whether I should chop this into smaller commits like:
* update `py-attrs`
* update `py-hatchling`
* update `py-jsonschema`
  * we might want to keep back the current version in other packages if we want to be on the _very safe side_ (see the summary below)
* update `py-bluepysnap`

# Summary of changes
## bluepysnap
`bluebrain/repo-bluebrain/packages/py-bluepysnap/package.py`
* `py-bluepysnap` `1.0.7` -> `2.0.1`
* [breaking changes](https://bluebrainsnap.readthedocs.io/en/stable/changelog.html#breaking-changes)
* previous version held:
  * `py-archngv`
  * `py-bluecellulab`
  * `py-connectome-manipulator`
* intentionally using the new version:
  * `brainbuilder`
  * `py-bbp-workflow`

## Added versions for existing packages as required by `bluepysnap` `2.0.1` 
`var/spack/repos/builtin/packages/py-attrs/package.py`
* `py-attrs` `22.1.0` -> `22.2.0`
* [no breaking changes](https://www.attrs.org/en/stable/changelog.html#id3)
  * other than for `py35` not being supported

`var/spack/repos/builtin/packages/py-hatchling/package.py`
* `py-hatchling` `1.10.0` -> `1.17.1`
* [no breaking changes](https://hatch.pypa.io/latest/history/hatchling/#hatchling-v1.17.1)
  * just fixes and additions

`var/spack/repos/builtin/packages/py-jsonschema/package.py`
* `py-jsonschema` `4.16.0` -> `4.19.2`
* ["should be backwards compatible"](https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst#v4180)
  * deprecations
  * fixes

## Added as dependencies:
`var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py`, required by
* `py-jsonschema`

`var/spack/repos/builtin/packages/py-trove-classifiers/package.py`, required by
* `py-hatchling`

`var/spack/repos/builtin/packages/py-referencing/package.py` required by
* `py-bluepysnap`
* `py-jsonschema`
* `py-jsonschema-specifications`

`var/spack/repos/builtin/packages/py-rpds-py/package.py` required by
* `py-referencing`
* `py-jsonschema`

File: var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py:1]: Are `builtin` changes coming from upstream?

Comment (joni-herttuainen) [reply var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py:1]: No, they are not. Should they be?

**Comment by joni-herttuainen** [comment]
We could (not necessarily "should") also rollback `bluepysnap` to use `jsonschema<=4.16`. Then we would not need to touch `builtin` (no updates/no extra packages).  But I guess we would need to address the deprecations in jsonschema at some point anyway, so it seems like a step back that does not feel good.

**EDIT:**
For reference, it would not be a huge [change](https://github.com/BlueBrain/snap/pull/232/commits/1d64d4ce3137a069ded4f1f75311116e566dc3a3) to rollback and reapply at some point.

File: var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-jsonschema-specifications/package.py:1]: They should be upstreamed if they are not. If we do another deployment, we'll throw away all the changes in `builtin`.

**Comment by joni-herttuainen** [comment]
I'll close this PR in favor of #2188 which introduces way less changes due to not having to update `py-jsonschema`.

**Pull request closed without merging**